vulkan_generator/handle_remapper/generator.py

```diff
@@ -0,0 +1,299 @@
+# Copyright (C) 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This is the entry point for Vulkan Code Generator"""
+
+import os
+
+from pathlib import Path
+from typing import List
+
+from textwrap import dedent
+from vulkan_parser import type_parser
+from codegen_utils import codegen
+
+def is_dispatchable(handle: str) -> bool:
+    return len(handle) < 10 # Not actually the correct criteria for whether a handle is dispatchable
+
+def handle_map_name(handle: str) -> str:
+    return f"""{handle[2:3].lower() + handle[3:]}HandleMap_"""
+
+def handle_count_map_name(handle: str) -> str:
+    return f"""{handle[2:3].lower() + handle[3:]}CountMap_"""
+
+def handle_add_name(handle: str) -> str:
+    return f"""add{handle[2:3].upper() + handle[3:]}Handle"""
+
+def handle_remove_name(handle: str) -> str:
+    return f"""remove{handle[2:3].upper() + handle[3:]}Handle"""
+
+def handle_remap_name(handle: str) -> str:
+    return f"""remap{handle[2:3].upper() + handle[3:]}Handle"""
+
+def dispatchable_handle_add_code(handle: str) -> str:
+    map_name = handle_map_name(handle)
+    return dedent(f"""
+        if({map_name}.find(captureHandle) != {map_name}.end()) throw HandleCollisionException();
+        {map_name}[captureHandle] = replayHandle;"""
+    )
+
+def dispatchable_handle_remove_code(handle: str) -> str:
+    map_name = handle_map_name(handle)
+    return dedent(f"""
+        if({map_name}.find(captureHandle) == {map_name}.end()) throw RemoveNonExistantHandleException();
+        {map_name}.erase(captureHandle);"""
+    )
+
+def dispatchable_handle_remap_code(handle: str) -> str:
```

Comment:
Consideration:

If you look at the function names, they are Class methods written without classes. Which leads encoding class name into functions.

Instead this could be much simpler:

```
class DispatchableHandleGenerator(HandleGenerator):
   handleName: str,

   def add_handle() -> str:
       # Definition

  # Other methods
  def generate():
       return add_handle() + remove_handle() + .....

generator = DispatchableHandleGenerator(handleName)
generator.generate()
```

Reply:
I think this makes the code more complex and adds a few more lines, but I've done it since you have strong feelings about this.

I haven't written a single function to generate all 3 function implementations since there's no reason to artificially join these separate functional pieces. One function with one purpose is neater.